### DISCUSSION TITLE
warning The use of this script is discouraged

### DISCUSSION THREAD

**Discussion by remco-pc** [doctrine/doctrine2#9958] [Support Questions]
### Bug Report

[Warning] The use of this script is discouraged. See https://www.doctrine-project.org/projects/doctrine-orm/en/current/reference/tools.html#doctrine-console for instructions on bootstrapping the console runner.

Version     2.12.3

#### Summary

I have used the console runner now and still get this warning.

My code:

```php
<?php
/**
 * @author          Remco van der Velde
 * @since           2022-07-31
 * @version         1.0
 * @license         MIT
 * @changeLog
 *     -    all
 */

use Doctrine\ORM\Tools\Console\ConsoleRunner;
use Doctrine\ORM\Tools\Console\EntityManagerProvider\SingleManagerProvider;

$dir = __DIR__;
$dir_vendor =
    $dir .
    DIRECTORY_SEPARATOR .
    'vendor' .
    DIRECTORY_SEPARATOR;

$autoload = $dir_vendor . 'autoload.php';
$autoload = require $autoload;
$config = new R3m\Io\Config(
    [
        'dir.vendor' => $dir_vendor
    ]
);
$app = new R3m\Io\App($autoload, $config);
echo R3m\Io\App::run($app);
$options = [];
$commands = [];
$entityManager = \R3m\Io\Module\Database::entityManager($app, $options);

ConsoleRunner::run(
    new SingleManagerProvider($entityManager),
    $commands
);
```

**Comment by remco-pc** [comment]
This is the new method written in the documentation, I think it's a bug.

**Reply by SenseException** [reply]
Can you please create a code example without using R3m? I don't understand where this warning is happening and what "this new method" is.

**Reply by remco-pc** [reply]
```php

ConsoleRunner::run(
    new SingleManagerProvider($entityManager),
    $commands
);
````
this piece of code is generating the message, hence i need an entity manager, its version is 2.13.1.
```
vendor/bin/doctrine
[Warning] The use of this script is discouraged. See https://www.doctrine-project.org/projects/doctrine-orm/en/current/reference/tools.html#doctrine-console for instructions on bootstrapping the console runner.
````

**Reply by SenseException** [reply]
True, but a lot of arguments aren't and that is why I'm asking to reproduce this error only with Doctrine classes.

**Reply by remco-pc** [reply]
here is a snippet without R3m and the same warning running:

```
vendor/bin/doctrine
````

```php
<?php
/**
 * @author          Remco van der Velde
 * @since           2022-07-31
 * @version         1.0
 * @license         MIT
 * @changeLog
 *     -    all
 */

use Doctrine\ORM\EntityManager;
use Doctrine\ORM\ORMSetup;
use Doctrine\ORM\Tools\Console\ConsoleRunner;
use Doctrine\ORM\Tools\Console\EntityManagerProvider\SingleManagerProvider;

$dir = __DIR__;
$dir_vendor =
    $dir .
    DIRECTORY_SEPARATOR .
    'vendor' .
    DIRECTORY_SEPARATOR;

$autoload = $dir_vendor . 'autoload.php';
$autoload = require $autoload;
$commands = [];
$entityManager = getEntityManager();

ConsoleRunner::run(
    new SingleManagerProvider($entityManager),
    $commands
);

/**
 * @throws \Doctrine\ORM\Exception\ORMException
 */
function getEntityManager(): EntityManager
{
    $connection = [];
    $connection['driver'] = 'pdo_mysql';
    $connection['host'] = '*';
    $connection['dbname'] = '*';
    $connection['user'] = '*';
    $connection['password'] = '*';
    $paths = [
        '/Application/src/Entity/'
    ];
    $proxyDir = null;
    $cache = null;
    $config = ORMSetup::createAnnotationMetadataConfiguration($paths, false, $proxyDir, $cache);
    return EntityManager::create($connection, $config);
}

```

**Reply by SenseException** [reply]
This is a relatively recent change to ORM. It actually discourage the usage of `vendor/bin/doctrine`, or more precisely the delivered console files of ORM, because most frameworks with ORM support come with their own implementation of console commands. In some standalone cases a ORM console is still needed though, but it can be set up with a few steps.

The [documentation](https://www.doctrine-project.org/projects/doctrine-orm/en/2.12/reference/tools.html#setting-up-the-console) is explaining how you can create your own console-file, which is actually what you did in parts. If you put

``` php
#!/usr/bin/env php
<?php
// ...
```

at the top of your file and make it executable with `chmod +x doctrine.php` (assuming your file is named doctrine.php and not cli-config.php) then you don't need `vendor/bin/doctrine` anymore. Since console files don't have a php suffix run `mv doctrine.php doctrine` and you can now execute it with `./doctrine`. Be sure to put it where other console files are, like in your own bin-directory.

**Reply by remco-pc** [reply]
i didn't had a /usr/bin/env php so i did this:
renamed cli-config.php to /Application/Bin/Doctrine.php  

and added : 

```
#!/bin/sh
_=doctrine php /Application/Bin/Doctrine.php "$@"
```
to /usr/bin/doctrine with chmod +x on /usr/bin/doctrine
